### PR TITLE
To validate the image_pull_secret in config file

### PR DESCRIPTION
**Pull request by vlella**
fix https://github.com/noironetworks/acc-provision/issues/579

### CODE REVIEW

**Comment by snaiksat** [comment]
@vlella thanks, can you please update the commit title and message?
Let's try to follow these guidelines:
https://chris.beams.io/posts/git-commit/
https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

File: provision/acc_provision/acc_provision.py [hunk 1]
```diff
@@ -1508,6 +1518,12 @@ def provision(args, apic_file, no_random):
         err("overlapping subnets found in configuration input file")
         return False
 
+    # Verify that image_pull_secret is a valid K8s secret name and not a YAML string
+    if "registry" in config.keys() and "image_pull_secret" in config["registry"]:
+        if not check_image_pull_secret(config):
+            err("The image_pull_secret is Invalid. It is supposed to be name of the k8s secret and it shouldn't have characters like =,+,!,~,@,& etc")
```
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:1524]: Rephrase this to say:
Invalid image_pull_secret value, it must be a valid DNS subdomain name.

File: provision/acc_provision/acc_provision.py [hunk 2]
```diff
@@ -1365,6 +1365,16 @@ def check_overlapping_subnets(config):
     return True
 
 
+def check_image_pull_secret(config):
+    # Check if the image_pull_secret is valid
+    image_pull_secret = config["registry"]["image_pull_secret"]
+    # This is the regex used by kubectl to validate objects names
+    pattern = "[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
```
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:1372]: Try to compare this with other examples that people might have posted for validating DNS subdomain names, and accordingly lets add a UT which validates, both, the positive and the negative cases.

**Review by snaiksat — CHANGES_REQUESTED** [review]

File: provision/acc_provision/acc_provision.py (hunk 1 above)
Comment (vlella) [reply provision/acc_provision/acc_provision.py]: Yes updated as requested. Thanks for the input @snaiksat

File: provision/acc_provision/acc_provision.py (hunk 2 above)
Comment (vlella) [reply provision/acc_provision/acc_provision.py:1372]: okay let me understand how to write a UT for validating DNS subdomain names then may be similarly try to mimic the work for our use case - image_pull_secret 

I guess the understanding is to add UT here - https://github.com/noironetworks/acc-provision/blob/master/provision/acc_provision/test_main.py
where 67 unit tests are already written for other work.
@snaiksat

Comment (snaiksat) [reply provision/acc_provision/acc_provision.py:1372]: To clarify, what I meant is compare the regex pattern we have used with the one others have used for DNS subdomain names to ensure we are not missing anything.

Comment (vlella) [reply provision/acc_provision/acc_provision.py:1372]: okay makes sense, thanks for clarifying! @snaiksat 

Regarding adding UT 
(67 existing ones + adding 1 new one for image_pull_secret) 
I guess with respect to this the understanding is right from my end?

**Review by snaiksat — APPROVED** [review]

**Review by snaiksat — APPROVED** [review]
